js/app/components/Payment/index.js

```diff
@@ -231,11 +134,56 @@ export default function(form, options) {
           break
         case 'edenred+card':
         case 'card':
-          handleCardPayment(cc, options, form, submitButton, savedPaymentMethod)
+          handleCardPayment(savedPaymentMethod)
           break
       }
     }
+  }
+
+  form.addEventListener('submit', async function(event) {
+
+    event.preventDefault()
+
+    setLoading(true)
+
+    //FIXME: only /order/payment route is tested to provide orderId; guest and account tokens
+
+    if (options.orderId) {
+      const httpClient = new window._auth.httpClient();
+
+      if (isGuest(options)) {
+        httpClient.setToken(options.orderAccessToken);
+      }
+
+      const hasAccount = window._auth && window._auth.isAuth
+
+      if (isGuest(options) || hasAccount) {
+        const validateItemRoute = window.Routing.generate("api_orders_validate_item", { id: options.orderId });
+        const result = await httpClient.get(validateItemRoute);
```

Comment:
Nice use of the _auth client :upside_down_face: 

As describe in that [page](https://antoinebagnaud.notion.site/Inject-auth-into-Symfony-view-0977a318f79f48dca329d7760e6db2be) you can deconstruct the return to check if there is any error during the request

```javascript
const httpClient = new window._auth.httpClient();
const { response, error } = await httpClient.post(url, payload);

if (error)
{
	// Do some error handling stuff.
	console.error(error);
	return;
}

// Use response object as you want.
// The response is a parsed json.
console.log(response);
```
And i think a good default behavior is to add a generic error message that simply say "Something went wrong", it's not perfect but it's better than nothing for the user :)

https://github.com/coopcycle/coopcycle-web/blob/cfad1ed79f0b23d8cbd579be32eab5393cb2da4e/js/app/dashboard/components/TaskReportIncidentModalContent.js#L66-L73


The notification came from the [antd library](https://4x.ant.design/components/notification/), the modal is sexy and pretty handy to use